service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/OrderRoutes.kt

```diff
@@ -0,0 +1,136 @@
+import org.thechance.service_restaurant.domain.utils.exceptions.INVALID_REQUEST_PARAMETER
+import org.thechance.service_restaurant.domain.utils.exceptions.MultiErrorException
+import org.thechance.service_restaurant.domain.utils.exceptions.NOT_FOUND
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.collections.set
+
+
+private val openingRestaurants: ConcurrentHashMap<String, RestaurantInfo> = ConcurrentHashMap()
+private val scope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
+
+fun Route.orderRoutes() {
+    val manageOrder: IManageOrderUseCase by inject()
+
+    route("/order") {
+
+        get("/{id}") {
+            val id = call.parameters["id"] ?: throw MultiErrorException(listOf(NOT_FOUND))
+            val result = manageOrder.getOrderById(orderId = id)
+            call.respond(HttpStatusCode.OK, result.toDto())
+        }
+
+        post("/{id}/status") {
+            val id = call.parameters["id"] ?: throw MultiErrorException(listOf(NOT_FOUND))
+            val status = call.receiveParameters()["status"]?.toInt() ?: throw MultiErrorException(
+                listOf(INVALID_REQUEST_PARAMETER)
+            )
+            val result = manageOrder.updateOrderStatus(
+                orderId = id,
+                state = OrderStatus.getOrderStatus(status)
+            )
+            call.respond(HttpStatusCode.OK, result)
+        }
+
+        get("/history/{id}") {
+            val id = call.parameters["id"] ?: throw MultiErrorException(listOf(NOT_FOUND))
+            val page = call.parameters["page"]?.toInt() ?: 1
+            val limit = call.parameters["limit"]?.toInt() ?: 10
+
+            val result = manageOrder.getOrdersHistory(restaurantId = id, page = page, limit = limit)
+            call.respond(HttpStatusCode.OK, result.map { it.toDto() })
+        }
+
+        get("/{restaurantId}/orders") {
+            val id = call.parameters["restaurantId"] ?: throw MultiErrorException(listOf(NOT_FOUND))
+            val result = manageOrder.getActiveOrdersByRestaurantId(restaurantId = id)
+            call.respond(HttpStatusCode.OK, result.map { it.toDto() })
+        }
+
+        webSocket("/{restaurantId}/{userId?}") {
+            val restaurantId = call.parameters["restaurantId"]?.trim().orEmpty()
+            val userId = call.parameters["userId"]?.trim().orEmpty()
+
+            if (userId.isEmpty()) {
+                openingRestaurants[restaurantId] = RestaurantInfo(owner = this, mutableListOf())
+
+                for (frame in incoming) return@webSocket
+
+            } else if (manageOrder.isRestaurantOpened(restaurantId)) {
+                openingRestaurants[restaurantId]?.users?.add(mutableMapOf(userId to this))
+                broadcast(
+                    receiveChannel = incoming,
+                    restaurantId = restaurantId,
+                    manageOrder = manageOrder
+                )
+            }
+        }
+
+    }
+}
+
+private suspend fun broadcast(
```

Comment:
i think this should be in different file !

Reply:
Done ✔